aea/skills/tasks.py

```diff
@@ -18,72 +18,102 @@
 
 """This module contains the classes for tasks."""
 import logging
+import multiprocessing
+import signal
 import threading
-from collections import deque
-from concurrent.futures import Executor, ThreadPoolExecutor
-from queue import Queue
-from threading import Thread
-from typing import Optional
-
-from aea.skills.base import Task
+from multiprocessing.pool import AsyncResult, Pool
+from typing import Any, Callable, Dict, Optional, Sequence, cast
 
 logger = logging.getLogger(__name__)
 
 
+def init_worker():
+    """
+    Initialize a worker.
+
+    Disable the SIGINT handler.
+    Related to a well-known bug: https://bugs.python.org/issue8296
+    """
+    signal.signal(signal.SIGINT, signal.SIG_IGN)
+
+
 class TaskManager:
     """A Task manager."""
 
-    def __init__(self, executor: Optional[Executor] = None):
+    def __init__(self, nb_workers: int = 5):
         """
         Initialize the task manager.
 
-        :param executor: the executor
+        :param nb_workers: the number of worker processes.
         """
-        self._executor = executor if executor is not None else ThreadPoolExecutor()
-
-        self.task_queue = Queue()  # type: Queue
-        self.futures = deque([])  # type: deque
-
-        self.stopped = True
-        self.thread = Thread(target=self.dispatch)
-        self.lock = threading.Lock()
-
-    def enqueue_task(self, task: Task) -> None:
+        self._nb_workers = nb_workers
+        self._pool = None  # type: Optional[Pool]
+        self._stopped = True
+        self._lock = threading.Lock()
+
+        self._task_enqueued_counter = 0
+        self._results_by_task_id = {}  # type: Dict[int, Any]
+
+    @property
+    def nb_workers(self) -> int:
+        """Get the number of workers."""
+        return self._nb_workers
+
+    def enqueue_task(
+        self, func: Callable, args: Sequence = (), kwds: Optional[Dict[str, Any]] = None
+    ) -> AsyncResult:
```

Comment:
this is the main function to be used at skill code level to enqueue tasks.

The `TaskManager` sees only callables, so it is not coupled to the `Task` class and so it solves a more general problem.

Reply:
it is still useful to have a wrapper, so the user can use it instead of interacting with the "bare" `multiprocessing.Pool`.

Reply:
Could also be mentioned in docs